09/Time.java

```diff
@@ -0,0 +1,50 @@
+import java.util.Random;
+
+public class Time {
+    private int hours;
+    private int minutes;
+
+    public Time(int hours, int minutes) {
+        if (hours > 0 && hours < 24) {
+            this.hours = hours;
+        } else {
+            //throw
+        }
+        if (minutes > 0 && minutes < 60) {
+            this.minutes = minutes;
+        } else {
+            //throw
+        }
+    }
+
+    public int getHours() {
+        return hours;
+    }
+
+    public int getMinutes() {
+        return minutes;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (!(obj instanceof Time)) {
+            return false;
+        }
+        Time other = (Time) obj;
+        return (other.hours == this.hours) && (other.minutes == this.minutes);
+    }
+
+    @Override
+    public String toString() {
+        return String.format("%d : %d", hours, minutes);
+
+    }
+
+    public Object Randomtime() {
```

Comment:
```suggestion
    public static Time Random() {
```
This has 3 parts:
1. The factory needs to be static - it belongs to the class, not each specific object
2. We want to give as much runtime type information as possible - why use `Object` instead of `Time`?
3. The "time" at the end of the name is redundant. The convention for factories is UpperCamelCase (unlike other Java functions that are in lowerCamelCase) - you did this well.